From 68682d0a868ee2a5378f5df2d17476fc7de31d4a Mon Sep 17 00:00:00 2001
From: Gary Lin <glin@suse.com>
Date: Fri, 31 Mar 2017 15:55:22 +0800
Subject: [PATCH 18/55] Cryptlib: Rename OpenSslSupport.h as CrtLibSupport.h

Edk2 renamed OpenSslSupport.h, so we have to follow the change.
Also merge some changes from edk2 CrtLibSupport.h

Signed-off-by: Gary Lin <glin@suse.com>
---
 .../{OpenSslSupport.h => CrtLibSupport.h}     | 84 +++++++++----------
 Cryptlib/Include/arpa/inet.h                  |  2 +-
 Cryptlib/Include/assert.h                     |  2 +-
 Cryptlib/Include/ctype.h                      |  2 +-
 Cryptlib/Include/dirent.h                     |  2 +-
 Cryptlib/Include/errno.h                      |  2 +-
 Cryptlib/Include/limits.h                     |  2 +-
 Cryptlib/Include/malloc.h                     |  2 +-
 Cryptlib/Include/math.h                       |  2 +-
 Cryptlib/Include/memory.h                     |  2 +-
 Cryptlib/Include/netdb.h                      |  2 +-
 Cryptlib/Include/netinet/in.h                 |  2 +-
 Cryptlib/Include/sgtty.h                      |  2 +-
 Cryptlib/Include/signal.h                     |  2 +-
 Cryptlib/Include/stdarg.h                     |  2 +-
 Cryptlib/Include/stddef.h                     |  2 +-
 Cryptlib/Include/stdio.h                      |  2 +-
 Cryptlib/Include/stdlib.h                     |  2 +-
 Cryptlib/Include/string.h                     |  2 +-
 Cryptlib/Include/strings.h                    |  2 +-
 Cryptlib/Include/sys/ioctl.h                  |  2 +-
 Cryptlib/Include/sys/param.h                  |  2 +-
 Cryptlib/Include/sys/socket.h                 |  2 +-
 Cryptlib/Include/sys/stat.h                   |  2 +-
 Cryptlib/Include/sys/time.h                   |  2 +-
 Cryptlib/Include/sys/times.h                  |  2 +-
 Cryptlib/Include/sys/types.h                  |  2 +-
 Cryptlib/Include/sys/un.h                     |  2 +-
 Cryptlib/Include/syslog.h                     |  2 +-
 Cryptlib/Include/time.h                       |  2 +-
 Cryptlib/Include/unistd.h                     |  2 +-
 Cryptlib/InternalCryptLib.h                   |  2 +-
 Cryptlib/SysCall/BaseMemAllocation.c          |  2 +-
 Cryptlib/SysCall/BaseStrings.c                |  2 +-
 Cryptlib/SysCall/CrtWrapper.c                 |  2 +-
 Cryptlib/SysCall/TimerWrapper.c               |  2 +-
 36 files changed, 77 insertions(+), 77 deletions(-)
 rename Cryptlib/Include/{OpenSslSupport.h => CrtLibSupport.h} (85%)

diff --git a/Cryptlib/Include/OpenSslSupport.h b/Cryptlib/Include/CrtLibSupport.h
similarity index 85%
rename from Cryptlib/Include/OpenSslSupport.h
rename to Cryptlib/Include/CrtLibSupport.h
index b38043c..279956d 100644
--- a/Cryptlib/Include/OpenSslSupport.h
+++ b/Cryptlib/Include/CrtLibSupport.h
@@ -1,7 +1,8 @@
 /** @file
-  Root include file to support building OpenSSL Crypto Library.
+  Root include file of C runtime library to support building the third-party
+  cryptographic library.
 
-Copyright (c) 2010 - 2011, Intel Corporation. All rights reserved.<BR>
+Copyright (c) 2010 - 2017, Intel Corporation. All rights reserved.<BR>
 This program and the accompanying materials
 are licensed and made available under the terms and conditions of the BSD License
 which accompanies this distribution.  The full text of the license may be found at
@@ -12,8 +13,8 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#ifndef __OPEN_SSL_SUPPORT_H__
-#define __OPEN_SSL_SUPPORT_H__
+#ifndef __CRT_LIB_SUPPORT_H__
+#define __CRT_LIB_SUPPORT_H__
 
 #include <efi.h>
 #include <efilib.h>
@@ -28,6 +29,9 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
  */
 #include <stddef.h>
 
+#define OPENSSLDIR ""
+#define ENGINESDIR ""
+
 #define CONST const
 
 //
@@ -169,27 +173,14 @@ typedef CHAR8 *VA_LIST;
 #endif
 
 //
-// #defines from EFI Application Toolkit required to buiild Open SSL
+// Definitions for global constants used by CRT library routines
 //
-#define ENOMEM       12               /* Cannot allocate memory */
 #define EINVAL       22               /* Invalid argument */
-#define BUFSIZ       1024             /* size of buffer used by setbuf */
-#define INT_MAX      2147483647       /* max value for an int */
-#define INT_MIN      (-2147483647-1)  /* min value for an int */
-#define LONG_MAX     2147483647L      /* max value for a long */
-#define LONG_MIN     (-2147483647-1)  /* min value for a long */
-#define ULONG_MAX    0xffffffff       /* max value for an unsigned long */
-#define LOG_DAEMON   (3<<3)           /* system daemons */
-#define LOG_EMERG    0                /* system is unusable */
-#define LOG_ALERT    1                /* action must be taken immediately */
-#define LOG_CRIT     2                /* critical conditions */
-#define LOG_ERR      3                /* error conditions */
-#define LOG_WARNING  4                /* warning conditions */
-#define LOG_NOTICE   5                /* normal but significant condition */
-#define LOG_INFO     6                /* informational */
-#define LOG_DEBUG    7                /* debug-level messages */
-#define LOG_PID      0x01             /* log the pid with each message */
-#define LOG_CONS     0x02             /* log on the console if errors in sending */
+#define INT_MAX      0x7FFFFFFF       /* Maximum (signed) int value */
+#define LONG_MAX     0X7FFFFFFFL      /* max value for a long */
+#define LONG_MIN     (-LONG_MAX-1)    /* min value for a long */
+#define ULONG_MAX    0xFFFFFFFF       /* Maximum unsigned long value */
+#define CHAR_BIT     8                /* Number of bits in a char */
 
 //
 // Macros from EFI Application Toolkit required to buiild Open SSL
@@ -204,25 +195,26 @@ typedef CHAR8 *VA_LIST;
 #endif
 
 //
-// Basic types from EFI Application Toolkit required to buiild Open SSL
+// Basic types mapping
 //
 typedef UINTN          size_t;
 typedef INTN           ssize_t;
+typedef INT32          time_t;
+typedef UINT8          __uint8_t;
+typedef UINT8          sa_family_t;
+typedef UINT32         uid_t;
+typedef UINT32         gid_t;
 typedef INT64          off_t;
 typedef UINT16         mode_t;
-typedef long           time_t;
 typedef unsigned long  clock_t;
-typedef UINT32         uid_t;
-typedef UINT32         gid_t;
 typedef UINT32         ino_t;
 typedef UINT32         dev_t;
 typedef UINT16         nlink_t;
 typedef int            pid_t;
 typedef void           *DIR;
-typedef void           __sighandler_t (int);
 
 //
-// Structures from EFI Application Toolkit required to buiild Open SSL
+// Structures Definitions
 //
 struct tm {
   int   tm_sec;     /* seconds after the minute [0-60] */
@@ -243,6 +235,12 @@ struct timeval {
   long tv_usec;     /* time value, in microseconds */
 };
 
+struct sockaddr {
+  __uint8_t    sa_len;       /* total length */
+  sa_family_t  sa_family;    /* address family */
+  char         sa_data[14];  /* actually longer; address value */
+};
+
 struct dirent {
   UINT32  d_fileno;         /* file number of entry */
   UINT16  d_reclen;         /* length of this record */
@@ -275,38 +273,37 @@ struct stat {
 };
 
 //
-// Externs from EFI Application Toolkit required to buiild Open SSL
+// Global variables
 //
 extern int errno;
 
 //
-// Function prototypes from EFI Application Toolkit required to buiild Open SSL
+// Function prototypes of CRT Library routines
 //
 void           *malloc     (size_t);
 void           *realloc    (void *, size_t);
 void           free        (void *);
-int            isdigit     (int);
-int            isspace     (int);
-int            tolower     (int);
-int            isupper     (int);
-int            isxdigit    (int);
-int            isalnum     (int);
 void           *memcpy     (void *, const void *, size_t);
-void           *memset     (void *, int, size_t);
 void           *memchr     (const void *, int, size_t);
 int            memcmp      (const void *, const void *, size_t);
 void           *memmove    (void *, const void *, size_t);
+void           *memset     (void *, int, size_t);
+int            isdigit     (int);
+int            isspace     (int);
+int            isxdigit    (int);
+int            isalnum     (int);
+int            isupper     (int);
+int            tolower     (int);
 int            strcmp      (const char *, const char *);
 int            strncmp     (const char *, const char *, size_t);
+int            strncasecmp (const char *, const char *, size_t);
+int            strcasecmp  (const char *, const char *);
 char           *strcpy     (char *, const char *);
 char           *strncpy    (char *, const char *, size_t);
 size_t         strlen      (const char *);
 char           *strcat     (char *, const char *);
 char           *strchr     (const char *, int);
-int            strcasecmp  (const char *, const char *);
-int            strncasecmp (const char *, const char *, size_t);
 char           *strncpy    (char *, const char *, size_t);
-int            strncmp     (const char *, const char *, size_t);
 char           *strrchr    (const char *, int);
 unsigned long  strtoul     (const char *, char **, int);
 long           strtol      (const char *, char **, int);
@@ -348,8 +345,11 @@ gid_t          getegid     (void);
 void           qsort       (void *, size_t, size_t, int (*)(const void *, const void *));
 char           *getenv     (const char *);
 void           exit        (int);
+#if defined(__GNUC__) && (__GNUC__ >= 2)
+void           abort       (void) __attribute__((__noreturn__));
+#else
 void           abort       (void);
-__sighandler_t *signal     (int, __sighandler_t *);
+#endif
 
 //
 // Global variables from EFI Application Toolkit required to buiild Open SSL
diff --git a/Cryptlib/Include/arpa/inet.h b/Cryptlib/Include/arpa/inet.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/arpa/inet.h
+++ b/Cryptlib/Include/arpa/inet.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/assert.h b/Cryptlib/Include/assert.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/assert.h
+++ b/Cryptlib/Include/assert.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/ctype.h b/Cryptlib/Include/ctype.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/ctype.h
+++ b/Cryptlib/Include/ctype.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/dirent.h b/Cryptlib/Include/dirent.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/dirent.h
+++ b/Cryptlib/Include/dirent.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/errno.h b/Cryptlib/Include/errno.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/errno.h
+++ b/Cryptlib/Include/errno.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/limits.h b/Cryptlib/Include/limits.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/limits.h
+++ b/Cryptlib/Include/limits.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/malloc.h b/Cryptlib/Include/malloc.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/malloc.h
+++ b/Cryptlib/Include/malloc.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/math.h b/Cryptlib/Include/math.h
index a21f554..b13508a 100644
--- a/Cryptlib/Include/math.h
+++ b/Cryptlib/Include/math.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/memory.h b/Cryptlib/Include/memory.h
index 092b3cd..4554616 100644
--- a/Cryptlib/Include/memory.h
+++ b/Cryptlib/Include/memory.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/netdb.h b/Cryptlib/Include/netdb.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/netdb.h
+++ b/Cryptlib/Include/netdb.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/netinet/in.h b/Cryptlib/Include/netinet/in.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/netinet/in.h
+++ b/Cryptlib/Include/netinet/in.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/sgtty.h b/Cryptlib/Include/sgtty.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sgtty.h
+++ b/Cryptlib/Include/sgtty.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/signal.h b/Cryptlib/Include/signal.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/signal.h
+++ b/Cryptlib/Include/signal.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/stdarg.h b/Cryptlib/Include/stdarg.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/stdarg.h
+++ b/Cryptlib/Include/stdarg.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/stddef.h b/Cryptlib/Include/stddef.h
index 8dfc36f..6f59a25 100644
--- a/Cryptlib/Include/stddef.h
+++ b/Cryptlib/Include/stddef.h
@@ -12,4 +12,4 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
diff --git a/Cryptlib/Include/stdio.h b/Cryptlib/Include/stdio.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/stdio.h
+++ b/Cryptlib/Include/stdio.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/stdlib.h b/Cryptlib/Include/stdlib.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/stdlib.h
+++ b/Cryptlib/Include/stdlib.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/string.h b/Cryptlib/Include/string.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/string.h
+++ b/Cryptlib/Include/string.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/strings.h b/Cryptlib/Include/strings.h
index 8dfc36f..6f59a25 100644
--- a/Cryptlib/Include/strings.h
+++ b/Cryptlib/Include/strings.h
@@ -12,4 +12,4 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
diff --git a/Cryptlib/Include/sys/ioctl.h b/Cryptlib/Include/sys/ioctl.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sys/ioctl.h
+++ b/Cryptlib/Include/sys/ioctl.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/sys/param.h b/Cryptlib/Include/sys/param.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sys/param.h
+++ b/Cryptlib/Include/sys/param.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/sys/socket.h b/Cryptlib/Include/sys/socket.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sys/socket.h
+++ b/Cryptlib/Include/sys/socket.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/sys/stat.h b/Cryptlib/Include/sys/stat.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sys/stat.h
+++ b/Cryptlib/Include/sys/stat.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/sys/time.h b/Cryptlib/Include/sys/time.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sys/time.h
+++ b/Cryptlib/Include/sys/time.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/sys/times.h b/Cryptlib/Include/sys/times.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sys/times.h
+++ b/Cryptlib/Include/sys/times.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/sys/types.h b/Cryptlib/Include/sys/types.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sys/types.h
+++ b/Cryptlib/Include/sys/types.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/sys/un.h b/Cryptlib/Include/sys/un.h
index ee07f6b..e95c19c 100644
--- a/Cryptlib/Include/sys/un.h
+++ b/Cryptlib/Include/sys/un.h
@@ -12,5 +12,5 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
diff --git a/Cryptlib/Include/syslog.h b/Cryptlib/Include/syslog.h
index 8dfc36f..6f59a25 100644
--- a/Cryptlib/Include/syslog.h
+++ b/Cryptlib/Include/syslog.h
@@ -12,4 +12,4 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
diff --git a/Cryptlib/Include/time.h b/Cryptlib/Include/time.h
index 8dfc36f..6f59a25 100644
--- a/Cryptlib/Include/time.h
+++ b/Cryptlib/Include/time.h
@@ -12,4 +12,4 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
diff --git a/Cryptlib/Include/unistd.h b/Cryptlib/Include/unistd.h
index 8dfc36f..6f59a25 100644
--- a/Cryptlib/Include/unistd.h
+++ b/Cryptlib/Include/unistd.h
@@ -12,4 +12,4 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
diff --git a/Cryptlib/InternalCryptLib.h b/Cryptlib/InternalCryptLib.h
index e9a4c20..538ce54 100644
--- a/Cryptlib/InternalCryptLib.h
+++ b/Cryptlib/InternalCryptLib.h
@@ -21,7 +21,7 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 #include "Library/DebugLib.h"
 #include "Library/BaseCryptLib.h"
 
-#include "OpenSslSupport.h"
+#include "CrtLibSupport.h"
 
 #include <openssl/opensslv.h>
 
diff --git a/Cryptlib/SysCall/BaseMemAllocation.c b/Cryptlib/SysCall/BaseMemAllocation.c
index 792b29e..aa0c3bf 100644
--- a/Cryptlib/SysCall/BaseMemAllocation.c
+++ b/Cryptlib/SysCall/BaseMemAllocation.c
@@ -13,7 +13,7 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
 //
 // -- Memory-Allocation Routines --
diff --git a/Cryptlib/SysCall/BaseStrings.c b/Cryptlib/SysCall/BaseStrings.c
index 252e6db..5c86ff6 100644
--- a/Cryptlib/SysCall/BaseStrings.c
+++ b/Cryptlib/SysCall/BaseStrings.c
@@ -1,4 +1,4 @@
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
 char *
 AsciiStrCat(char *Destination, char *Source)
diff --git a/Cryptlib/SysCall/CrtWrapper.c b/Cryptlib/SysCall/CrtWrapper.c
index 698e1ee..0b61f65 100644
--- a/Cryptlib/SysCall/CrtWrapper.c
+++ b/Cryptlib/SysCall/CrtWrapper.c
@@ -13,7 +13,7 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
 int errno = 0;
 
diff --git a/Cryptlib/SysCall/TimerWrapper.c b/Cryptlib/SysCall/TimerWrapper.c
index 27ac44a..b302916 100644
--- a/Cryptlib/SysCall/TimerWrapper.c
+++ b/Cryptlib/SysCall/TimerWrapper.c
@@ -13,7 +13,7 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 
 **/
 
-#include <OpenSslSupport.h>
+#include <CrtLibSupport.h>
 
 //
 // -- Time Management Routines --
-- 
2.17.1

